src/handle_append_entries.cxx

```diff
@@ -338,6 +338,8 @@ ptr<resp_msg> raft_server::handle_append_entries(req_msg& req)
         }
     }
 
+    leader_ = req.get_src();
```

Comment:
We should not change `leader_` here before checking whether the given message is valid.